scripts/version

```diff
@@ -2,34 +2,57 @@
 
 TREE_STATE=clean
 COMMIT=${COMMIT:-${DRONE_COMMIT:-${GITHUB_SHA:-unknown}}}
-if [[ "$GITHUB_REF" = refs/tags/* ]]; then
-    GITHUB_TAG=${GITHUB_REF#refs/tags/}
-fi
 TAG=${TAG:-${DRONE_TAG:-$GITHUB_TAG}}
 
 if [ -d .git ]; then
-    if [ -z "$TAG" ]; then
-        TAG=$(git tag -l --contains HEAD | head -n 1)
-    fi
     if [ -n "$(git status --porcelain --untracked-files=no)" ]; then
-        DIRTY="-dirty"
+        DIRTY="dirty"
         TREE_STATE=dirty
     fi
 
+    if [[ "$TREE_STATE" == "clean" && -z "$TAG" ]]; then
+        TAG=$(git tag -l --contains HEAD | head -n 1) # this is going to not work if you have multiple tags pointing to the same commit
```

Comment:
Why not use `git describe --tags` here? IT has a well defined format that can be parsed if needed.

Reply:
Additionally, `git tag -l --contains HEAD` will be empty for PR builds. Does this break assumptions? (Given the nature of this repo it may not matter)

Reply:
This is OK, as instead the version will get generated with the commit hash in it (line 28)

Reply:
As for not using `git describe --tags`, I took this original inspiration of this from another rancher repo and kept with that convention of using `git tag -l --contains HEAD`